dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -318,6 +357,82 @@ public CultureInfo Culture
     }
     #endregion
 
+    #region Internal Filtering
+
+    [Experimental("SKEXP0005")]
+    internal FunctionInvokingContext? OnFunctionInvokingFilter(KernelFunction function, KernelArguments arguments)
+    {
+        FunctionInvokingContext? context = null;
+
+        if (this._functionFilters is { Count: > 0 })
+        {
+            context = new(function, arguments);
+
+            foreach (var filter in this._functionFilters)
```

Comment:
Nit: this week allocate an enumerator. You'll avoid that if you instead use a for loop.